web/app.js

```diff
@@ -3125,6 +3121,10 @@ function onKeyDown(evt) {
   if (handled) {
     evt.preventDefault();
   }
+
+  if (this.editorUndoBar?.isOpen && evt.keyCode !== 9 && evt.keyCode !== 16) {
+    this.editorUndoBar?.hide();
+  }
```

Comment:
Placing this code at the end might miss some cases, please note the early return here: https://github.com/mozilla/pdf.js/blob/bde36f28bee3a01495375ae88d6ac2e3bf217c74/web/app.js#L2898-L2904